flink-core/src/main/java/org/apache/flink/util/ClassLoaderWithErrorHandler.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.util;
+
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.function.Consumer;
+
+/**
+ * This class loader accepts a custom handler if an exception occurs in {@link #loadClass(String, boolean)}.
+ */
+public abstract class ClassLoaderWithErrorHandler extends URLClassLoader {
+	public static final Consumer<Throwable> EMPTY_EXCEPTION_HANDLER = classLoadingException -> {};
+
+	private final Consumer<Throwable> classLoadingExceptionHandler;
+
+	protected ClassLoaderWithErrorHandler(URL[] urls, ClassLoader parent) {
+		this(urls, parent, EMPTY_EXCEPTION_HANDLER);
+	}
+
+	protected ClassLoaderWithErrorHandler(
+			URL[] urls,
+			ClassLoader parent,
+			Consumer<Throwable> classLoadingExceptionHandler) {
+		super(urls, parent);
+		this.classLoadingExceptionHandler = classLoadingExceptionHandler;
+	}
+
+	@SuppressWarnings("FinalMethod")
```

Comment:
Why do we need this suppression here? Is it because we make the protected method `final`?

Reply:
So far we have never used this annotation in the Flink code base.

Reply:
This is for IDE. I can remove it if it is not a common thing.